### ISSUE TITLE
Allow extending readiness timeouts

### ISSUE THREAD

**Issue by novoj** [FgForrest/evitaDB#700]
For pod operations, you typically implement liveness and readiness probes. In evitaDB, for example, I have them defined as follows:

```
HEALTHCHECK --start-period=5m --interval=10s --timeout=2s --retries=3 \
    CMD curl -f http://localhost:$SYSTEM_API_PORT/system/liveness || exit 1
```

This means querying the specific service every 10 seconds. If it doesn't respond within 2 seconds or returns an error—and this happens 3 times in a row—the pod is considered unhealthy and will be restarted.

Within the liveness check, I perform a "self-check," which involves additional HTTP requests to itself (across all open APIs) with a timeout of 1 second for each.

## The Problem:

On low-quality hosting services, the latency is highly variable. Usually, responses come back in low tens to hundreds of milliseconds. However, occasionally it stalls and responds only after 4 seconds. The deployed solution will experience similar latencies, but it's more concerning here because it triggers a pod restart.

Another issue is that the timeout in the Dockerfile cannot be configured, so it has to be very conservative.

## Proposed Solution:

Increase Dockerfile Timeout: Set the timeout in the Dockerfile to something like 15 seconds.
Configurable Internal Timeout: Implement a configurable timeout for the liveness service internally, with a default of 500ms. This can be extended for specific installations—for Contabo, for example, to 5 seconds, which appears to be a stable timeout.
Parallelize Checks: Parallelize the health checks since multiple HTTP calls occur (approximately 4). This prevents the 5-second response times from adding up and exceeding the 15-second limit at the liveness service level.
By making these adjustments, we can accommodate the variable latencies of cheaper hosting services without unnecessarily restarting pods due to transient delays.

**Comment by novoj** [comment]
Readiness probes use the `requestTimeout` settings from the ApiOptions and are checked in parallel in a separate thread pool.
